datasets/librispeech.py

```diff
@@ -0,0 +1,200 @@
+            "test-clean",
+            "test-other",
+            "train-clean-100",
+            "train-clean-360",
+            "train-other-500",
+        ]
+
+        # this is just needed to have the extraction target available
+        self._out_parent_folder = self.output_path(".", directory=True)
+
+        self.out_corpus_folder = self.output_path("LibriSpeech/%s" % self.corpus_key)
+
+    def tasks(self):
+        yield Task("run", mini_task=True)
+
+    def run(self):
+        subprocess.check_call(
+            ["wget", "https://www.openslr.org/resources/12/md5sum.txt"]
+        )
+        subprocess.check_call(
+            ["wget", "https://www.openslr.org/resources/12/%s.tar.gz" % self.corpus_key]
+        )
+
+        with open("md5sum.txt", "rt") as md5_in, open(
+            "md5sum-%s.txt" % self.corpus_key, "wt"
+        ) as md5_out:
+            for line in md5_in:
+                split = line.strip().split(" ")
+                if split[-1].split(".")[0] == self.corpus_key:
+                    md5_out.write(line)
+                    break
+
+        checksum_result = subprocess.check_output(
+            ["md5sum", "-c", "md5sum-%s.txt" % self.corpus_key]
+        )  #  type: bytes
+        assert b"OK" in checksum_result
+
+        subprocess.check_call(
+            [
+                "tar",
+                "-xf",
+                "%s.tar.gz" % self.corpus_key,
+                "-C",
+                self._out_parent_folder.get_path(),
+            ]
+        )
+
+
+class DownloadLibriSpeechMetadataJob(DownloadLibriSpeechCorpusJob):
+    """
+    Downloads the metadata file and checks for md5sum integrity
+
+    Defines outputs for "SPEAKERS.TXT, CHAPTERS.TXT and BOOKS.TXT"
+    """
+
+    # noinspection PyMissingConstructor
+    def __init__(self):
+        self.corpus_key = "raw-metadata"
+
+        # this is just needed to have the extraction target available
+        self._out_parent_folder = self.output_path(".", directory=True)
+
+        self.out_speakers = self.output_path("LibriSpeech/SPEAKERS.TXT")
+        self.out_chapters = self.output_path("LibriSpeech/CHAPTERS.TXT")
+        self.out_books = self.output_path("LibriSpeech/BOOKS.TXT")
+
+
+class LibriSpeechCreateBlissJob(Job):
+    """
+    Creates a Bliss corpus from a LibriSpeech corpus folder using the speaker information in addition
+
+    Outputs a single bliss .xml.gz file
+    """
+
+    def __init__(self, corpus_folder, speaker_metadata):
+        """
+        Generate a bliss xml from  the LibriSpeech corpus.
+
+        :param Path corpus_folder: Path to a LibriSpeech corpus folder
+        :param Path speaker_metadata: Path to SPEAKER.TXT file from the MetdataJob (out_speakers)
+        """
+        self.corpus_folder = corpus_folder
+        self.speaker_metadata = speaker_metadata
+
+        self.out_corpus = self.output_path("corpus.xml.gz")
+
+        self._speakers = {}  # dict(key: id, value: [sex, subset, min, name]
```

Comment:
Why are speakers lists and transcripts dicts? It would be more consistent to use the same data-structure here.

Reply:
I did not write the original code, but I think the reason is that the speakers have specific key and need to be accessed with that, while the transcript "list" has no specific key attribute and just needs to be iterated. I do not really see a sense in changing that.